drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -294,107 +322,81 @@ def studio_submit(self, submissions, suffix=''):
         }
 
     @XBlock.json_handler
-    def do_attempt(self, attempt, suffix=''):
-        item = self._get_item_definition(attempt['val'])
+    def drop_item(self, item_attempt, suffix=''):
+        """
+        Handles dropping item into a zone.
+        """
+        self._validate_drop_item(item_attempt)
 
-        state = None
-        zone = None
-        feedback = item['feedback']['incorrect']
-        overall_feedback = None
-        is_correct = False
-
-        if self._is_attempt_correct(attempt):  # Student placed item in a correct zone
-            is_correct = True
-            feedback = item['feedback']['correct']
-            state = {
-                'zone': attempt['zone'],
-                'correct': True,
-                'x_percent': attempt['x_percent'],
-                'y_percent': attempt['y_percent'],
-            }
-
-        if state:
-            self.item_state[str(item['id'])] = state
-            zone = self._get_zone_by_uid(state['zone'])
+        if self.mode == self.ASSESSMENT_MODE:
+            return self._drop_item_assessment(item_attempt)
+        elif self.mode == self.STANDARD_MODE:
+            return self._drop_item_standard(item_attempt)
         else:
-            zone = self._get_zone_by_uid(attempt['zone'])
-        if not zone:
-            raise JsonHandlerError(400, "Item zone data invalid.")
+            raise JsonHandlerError(
+                500,
+                self.i18n_service.gettext("Unknown DnDv2 mode {mode} - course is misconfigured").format(self.mode)
+            )
 
-        if self._is_finished():
-            overall_feedback = self.data['feedback']['finish']
+    @XBlock.json_handler
+    def do_attempt(self, data, suffix=''):
+        """
+        Checks submitted solution and returns feedback.
 
-        # don't publish the grade if the student has already completed the problem
-        if not self.completed:
-            if self._is_finished():
-                self.completed = True
-            try:
-                self.runtime.publish(self, 'grade', {
-                    'value': self._get_grade(),
-                    'max_value': self.weight,
-                })
-            except NotImplementedError:
-                # Note, this publish method is unimplemented in Studio runtimes,
-                # so we have to figure that we're running in Studio for now
-                pass
+        Raises 400 error in standard mode.
```

Comment:
Nit: might also be worth mentioning that it raises a 409 error if max attempts are exhausted.